src/v/kafka/group_probe.h

```diff
@@ -44,11 +52,86 @@ class group_offset_probe {
             [this] { return _offset; },
             sm::description("Group topic partition offset"),
             labels)});
+
+        auto ns_label = sm::label("namespace");
+        labels.push_back(ns_label(tp_ns.ns()));
+
+        _public_metrics.add_group(
+          prometheus_sanitize::metrics_name("kafka:consumer:group"),
+          {
+            sm::make_gauge(
+            "committed_offset",
+            [this] { return _offset; },
+            sm::description("Consumer group comitted offset"),
+            labels)
+            .aggregate({sm::shard_label})
+          });
     }
 
 private:
     model::offset& _offset;
     ss::metrics::metric_groups _metrics;
+    ss::metrics::metric_groups _public_metrics;
+};
+
+
+template <typename KeyType, typename ValType>
+class group_probe {
+  using member_map = absl::node_hash_map<kafka::member_id, member_ptr>;
+  using static_member_map = absl::node_hash_map<kafka::group_instance_id, kafka::member_id>;
+
+public:
+    explicit group_probe(member_map& members, static_member_map& static_members) noexcept
+      : _members(members)
+      , _static_members(static_members)
+      , _public_metrics(ssx::public_metrics_handle) {}
+
+    void setup_metrics(const kafka::group_id& group_id, absl::node_hash_map<KeyType,ValType>& offsets) {
+        namespace sm = ss::metrics;
+
+        if (config::shard_local_cfg().disable_metrics()) {
+            return;
+        }
+
+        auto group_label = sm::label("group");
+
+        std::vector<sm::label_instance> labels{
+          group_label(group_id())};
+
+        _public_metrics.add_group(
+          prometheus_sanitize::metrics_name("kafka:consumer:group"),
+          {
+            sm::make_gauge(
+            "consumers",
+            [this] { 
+              return _members.size() + _static_members.size();
+            },
+            sm::description("Number of consumers in a group"),
+            labels)
+            .aggregate({sm::shard_label}),
+
+            sm::make_gauge(
+            "topics",
+            [this, &offsets] {
+              // Capture list of unique topic partitions
+              // by storing them in a set
+              for (auto &elem: offsets) {
+                _topics.insert(elem.first);
+              }
+
+              return _topics.size();
```

Comment:
I think this is what we want:
```suggestion
              return _offsets.size();
```
Since the key is the same for the underlying map and the set.

But holding topics over time means that we're counting every topic that was ever registered, rather than the current view (accounting for removed topics). Is that what you intended?

Reply:
> Since the key is the same for the underlying map and the set.

Ah right, there are not duplicate keys in the offset map. I must've been tired when writing that in. I'll make the change.

> But holding topics over time means that we're counting every topic that was ever registered, rather than the current view (accounting for removed topics). Is that what you intended?

No that is not my intention, I also thought about this overnight and I'll remove the topics container in the next push.

Reply:
Done. Changes were pushed to a separate PR at https://github.com/redpanda-data/redpanda/pull/5349